src/Materials/Node/Blocks/Dual/textureBlock.ts

```diff
@@ -373,13 +373,17 @@ export class TextureBlock extends NodeMaterialBlock {
 
     private _generateConversionCode(state: NodeMaterialBuildState, output: NodeMaterialConnectionPoint, swizzle: string): void {
         if (swizzle !== 'a') { // no conversion if the output is "a" (alpha)
-            state.compilationString += `#ifdef ${this._linearDefineName}\r\n`;
-            state.compilationString += `${output.associatedVariableName} = toGammaSpace(${output.associatedVariableName});\r\n`;
-            state.compilationString += `#endif\r\n`;
+            if (!this.texture || !this.texture.gammaSpace) {
+                state.compilationString += `#ifdef ${this._linearDefineName}\r\n`;

```

Comment:
Template literals can come in handy here
```suggestion
                state.compilationString += `#ifdef ${this._linearDefineName}
${output.associatedVariableName} = toGammaSpace(${output.associatedVariableName});
#endif
`;
```